checkbox-provider-ce-oem/bin/cold_reboot_by_pdu.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+ip=""
+outlet_port=""
+pdu_type=""
+reboot_duration=""
+
+cold_reboot_apc() {
+    echo "Turn off the power of DUT by APC network PDU.."
+    echo "System should be back after around $reboot_duration seconds.."
+    echo ""
+
+    # change the duration of power reboot
+    oid="1.3.6.1.4.1.318.1.1.4.5.2.1.5.$outlet_port"
+    i=0
+    result="fail"
+    while [ $i -le 3 ]
+    do
+        echo "## change the duration of power reboot"
+        snmpset -v1 -c private "$ip" "$oid" integer "$reboot_duration" > /dev/null
+        ret="$?"
+        if [ "$ret" == 0 ]; then
+            echo "changed the duration successfully!"
+            result="pass"
+            break
+        fi
+        sleep 1
+        i=$((i + 1))
+    done
+    if [ $result == "fail" ]; then
+        echo "## failed to change the duration of power reboot"  >&2
+        exit 1
+    fi
+
+    # power cycle a outlet with delay
+    echo ""
+    oid="1.3.6.1.4.1.318.1.1.4.4.2.1.3.$outlet_port"
+    i=0
+    result="fail"
+    while [ $i -le 3 ]
+    do
+        echo "## trigger a cold reboot by network PDU.."
+        # 1: power on, 2: power off, 3: power cycle
+        snmpset -v1 -c private "$ip" "$oid" integer 3 > /dev/null
+        ret="$?"
+        if [ "$ret" == 0 ]; then
+            echo "trigger a cold reboot successfully!"
+            result="pass"
+            break
+        fi
+        sleep 1
+        i=$((i + 1))
+    done
+    if [ $result == "fail" ]; then
+        echo "## failed to trigger a cold reboot" >&2
+        exit 1
+    fi
+}
+
+
+main() {
+    ip=$(echo $pdu_var | awk -F ':' '{print $1}')
+    outlet_port=$(echo $pdu_var | awk -F ':' '{print $2}')
+
+    if [[ -z "$ip" || -z "$outlet_port" ]]; then
+        echo -e "Error: PDU variable format is unexpected!\n"
+        help_function
+        exit 1
+    fi
+
+    if [ "$pdu_type" == "apc" ]; then
+        cold_reboot_apc
+    fi
+}
+
+help_function() {
+    echo "This script is uses for trigger a cold reboot test by network PDU"
+    echo
+    echo "Usage: cold_reboot_by_pdu.sh -t type -p pud-ip:outlet-port"
+    echo -e "\t-t    Network PDU type. [apc]"
+    echo -e "\t-p    Network PDU IP and outlet port. e.g. 10.120.33.44:2"
+    echo -e "\t-d    Reboot delay duration. Default delay is 60 (seconds)"
+}
+
+while getopts "t:p:d:" opt; do
+    case "$opt" in
+        t) pdu_type="$OPTARG" ;;
+        p) pdu_var="$OPTARG" ;;
+        d) reboot_duration="$OPTARG" ;;
+        ?) help_function ;;
+    esac
+done
+
+if  [[ "$pdu_type" != "apc" ]]; then
+    echo -e "Error: Network PDU type is not supported!\n"
+    help_function
+    exit 1
+fi
```

Comment:
Seems this part of the code is not mandatory to check the type of PDU is APC or not. Maybe we could check if only the variable is given or not here. And put an exception in the section of main to check `$pdu_type`.
> if [ "$pdu_type" == "apc" ]; then
       cold_reboot_apc
   fi